src/rpc/misc.cpp

```diff
@@ -712,8 +755,20 @@ static void getHeightRange(const UniValue& params, int& start, int& end)
         if (!startValue.isNull() && !endValue.isNull()) {
             start = startValue.get_int();
             end = endValue.get_int();
+            if (start <= 0 || end <= 0) {
+                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY,
+                    "Start and end are expected to be greater than zero");
+            }
+            if (end < start) {
+                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY,
+                    "End value is expected to be greater than start");
+            }
         }
     }
+
+    if (start > chainActive.Height() || end > chainActive.Height()) {
```

Comment:
`chainActive` requires a lock on `cs_main`. Either grab the lock here, add an `AssertLockHeld(cs_main)` to the start of this function, or move this check out to the individual RPC methods. Ideally whichever option minimises the necessary locking period, and keeps the information gathered from the various locked sections consistent.